### PR TITLE
feat(react-native): CallControls customization

### PR DESCRIPTION
**Pull request by khushal87**
This PR focuses on Call control customization. The sub-components are exported and are mostly on par with the naming convention and pattern of React SDK.

Integrators can use the component directly to build their own Call controls view. We also provide a default Call controls view that can be used if integrators do not want any customization.

### CODE REVIEW

File: packages/react-native-sdk/src/components/call/CallControls/CallControls.tsx
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/CallControls.tsx:1]: wow looks so much neater now 👏

File: packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { theme } from '../../../theme';
+import { Phone } from '../../../icons';
+
+/**
+ * The props for the Accept Call button.
+ */
+type AcceptCallButtonProps = {
+  /**
+   * Handler to be called when the accept call button is pressed.
+   * @returns void
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx:13]:
```suggestion
```

for uniformity

File: packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * Button to accept a call.
+ *
+ * Mostly calls call.join() internally.
+ */
+export const AcceptCallButton = ({
+  onPressHandler,
+  onAcceptHandler,
+}: AcceptCallButtonProps) => {
+  const call = useCall();
+  const acceptCallHandler = useCallback(async () => {
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx:32]: we can remove this useCallback wrapper, unnecessary here

File: packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+type AcceptCallButtonProps = {
+  /**
+   * Handler to be called when the accept call button is pressed.
+   * @returns void
+   */
+  onPressHandler?: () => void;
+  /**
+   *  Handler to be called after the incoming call is accepted.
+   */
+  onAcceptHandler?: () => void;
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx:19]: this usecase seems new to me.. two handlers and one for after and one for complete

can we remove the `onAcceptHandler` here

File: packages/react-native-sdk/src/components/call/CallControls/ChatButton.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+export type ChatButtonProps = {
+  /**
+   * Handler to be called when the chat button is pressed.
+   * @returns void
+   */
+  onPressHandler?: () => void;
+  /**
+   * The unread message indicator to be displayed above on the Chat button.
+   */
+  unreadBadgeCountIndicator?: number;
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/ChatButton.tsx:21]:
```suggestion
  unreadBadgeCount?: number;
```

more easier to understand in my opinion, just an optional suggestion

File: packages/react-native-sdk/src/components/call/CallControls/HangupCallButton.tsx [hunk 5]
```diff
@@ -16,15 +16,24 @@ export type HangUpCallButtonProps = {
    * @returns void
    */
   onPressHandler?: () => void;
-
+  /**
+   * Handler to be called when the call is hanged up.
+   */
+  onHangUpCall?: () => void;
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/HangupCallButton.tsx:22]: I see the usecase but its confusing with two handlers..

I don't think other SDKs have it, can you please check and align

Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/HangupCallButton.tsx:22]: React has it named as `onLeave`

Comment (santhoshvai) [reply packages/react-native-sdk/src/components/call/CallControls/HangupCallButton.tsx:22]: if it makes it easy to have it, then we can add a clear comment that if onPress is passed the other handler is not invoked

File: packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx (hunk 3 above)
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx:19]: React has it, that is why I added it for uniformity. Integrator can display there logic on what should happen after accepting the call, without overriding the entire handler.

File: packages/react-native-sdk/src/components/call/CallControls/ChatButton.tsx (hunk 4 above)
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/ChatButton.tsx]: Done ✅

File: packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx (hunk 2 above)
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx]: Done ✅

File: packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx (hunk 1 above)
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/AcceptCallButton.tsx]: Done ✅

File: packages/react-native-sdk/src/components/call/CallControls/CallControls.tsx
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/CallControls.tsx:1]: Yes 🤩

File: packages/react-native-sdk/src/components/call/CallControls/HangupCallButton.tsx (hunk 5 above)
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/HangupCallButton.tsx:22]: Done ✅

File: packages/react-native-sdk/src/components/call/CallControls/RejectCallButton.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      }
+      await call?.leave({ reject: true });
+      if (onRejectHandler) {
+        onRejectHandler();
+      }
+    } catch (error) {
+      console.log('Error rejecting Call', error);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [call]);
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/RejectCallButton.tsx:54]: another unnecessary usecallback

File: packages/react-native-sdk/src/components/call/CallControls/ReactionButton.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+ * Button to display the list of Reactions supported in the call.
+ * On press, it opens a view that can be used to send Reaction.
+ */
+export const ReactionButton = ({ onPressHandler }: ReactionButtonProps) => {
+  const [showReactionsPicker, setShowReactionsPicker] =
+    useState<boolean>(false);
+  const [reactionsButtonLayoutRectangle, setReactionsButtonLayoutRectangle] =
+    useState<LayoutRectangle>();
+
+  // This is for the reaction popup
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/call/CallControls/ReactionButton.tsx:32]:
```suggestion
  // This is for the reaction picker
```

File: packages/react-native-sdk/src/components/call/CallControls/RejectCallButton.tsx (hunk 6 above)
Comment (khushal87) [reply packages/react-native-sdk/src/components/call/CallControls/RejectCallButton.tsx]: Removed

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**